[CI] EsqlClientYamlIT class failing

**Build Scans:**
- [elasticsearch-periodic #7417 / 8.11.4_bwc](https://gradle-enterprise.elastic.co/s/vjxc2fgavex54)
- [elasticsearch-periodic #7417 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/eqoxqtpjagzus)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.11.4#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/160_union_types/load two indices with incorrect conversion function, TO_LONG instead of TO_IP}" -Dtests.seed=516A5A7613A6660B -Dtests.bwc=true -Dtests.locale=gsw-CH -Dtests.timezone=Mexico/BajaNorte -Druntime.java=24
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: circuit breakers not reset to 0Expected a map containing
estimated_size_in_bytes: expected <0> but was <368>
         estimated_size: expected "0b" but was "368b"
               overhead: <1.0> unexpected but ok
    limit_size_in_bytes: <322122547> unexpected but ok
             limit_size: "307.1mb" unexpected but ok
                tripped: <0> unexpected but ok
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in class org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
- [8.x] 2 failures in class org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT (1.3% fail rate in 158 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.11.4#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/61_enrich_ip/IP strings}" -Dtests.seed=516A5A7613A6660B -Dtests.bwc=true -Dtests.locale=gsw-CH -Dtests.timezone=Mexico/BajaNorte -Druntime.java=24

EsqlClientYamlIT > test {p0=esql/61_enrich_ip/IP strings} FAILED
    java.lang.AssertionError: Failure at [esql/61_enrich_ip:112]: expected [2xx] status code but api [esql.query] returned [500 Internal Server Error] [---
    error:
      root_cause:
      - type: "null_pointer_exception"
        reason: "Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.noText()\"\
          \ because \"inputDataType\" is null"
        stack_trace: "org.elasticsearch.ElasticsearchException$1: Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.noText()\"\
          \ because \"inputDataType\" is null\n\tat org.elasticsearch.server@8.11.4/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:670)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:598)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.rest.RestResponse.build(RestResponse.java:186)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:134)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:112)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:55)\n\
          \tat org.elasticsearch.xpack.esql.action.EsqlResponseListener.lambda$wrapWithLogging$1(EsqlResponseListener.java:165)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:185)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.rest.action.RestCancellableNodeClient$1.onFailure(RestCancellableNodeClient.java:96)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.client.internal.node.NodeClient$SafelyWrappedActionListener.onFailure(NodeClient.java:171)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:215)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$execute$1(ComputeService.java:135)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:186)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:191)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:308)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListener$4.onFailure(ActionListener.java:336)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListener$4.onFailure(ActionListener.java:336)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:127)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:117)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:69)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:116)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.Releasables$3.close(Releasables.java:142)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:310)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:230)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:275)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:173)\n\
          \tat org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler.closeInternal(ExchangeSourceHandler.java:246)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:69)\n\
          \tat org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$PendingInstances.finishInstance(ExchangeSourceHandler.java:267)\n\
          \tat org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$RemoteSinkFetcher.onSinkComplete(ExchangeSourceHandler.java:208)\n\
          \tat org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$RemoteSinkFetcher.lambda$fetchPage$1(ExchangeSourceHandler.java:165)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:177)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:95)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:310)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:230)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:259)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:173)\n\
          \tat org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$runComputeOnRemoteNodes$8(ComputeService.java:221)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:117)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:69)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:116)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.Releasables$3.close(Releasables.java:142)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.IOUtils.close(IOUtils.java:71)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.Releasables.close(Releasables.java:33)\n\
          \tat org.elasticsearch.base@8.11.4/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:57)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:49)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:271)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListener$4.onFailure(ActionListener.java:336)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:54)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleException(TransportService.java:1719)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1436)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1572)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.transport.TransportService$DirectResponseChannel$2.doRun(TransportService.java:1552)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)\n\
          \tat org.elasticsearch.server@8.11.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
          \tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\
          \tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\
          \tat java.base/java.lang.Thread.run(Thread.java:1583)\nCaused by: java.lang.NullPointerException:\
          \ Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.noText()\"\
          \ because \"inputDataType\" is null\n\tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.rangeTypesCompatible(EnrichLookupService.java:139)\n\
          \tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.validateTypes(EnrichLookupService.java:128)\n\
          \tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:103)\n\
          \tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:63)\n\
          \tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService.doLookup(AbstractLookupService.java:302)\n\
          \tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:440)\n\
          \tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:435)\n\
          \tat org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:76)\n\
          \tat org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:289)\n\
          \tat org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:302)\n\
          \tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)\n\
          \tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\
          \tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)\n\
          \tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)\n\
          \tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\
          \tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)\n\
          \tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)\n\
          \tat java.lang.Thread.run(Thread.java:1447)\n"
      type: "null_pointer_exception"
      reason: "Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.noText()\"\
        \ because \"inputDataType\" is null"
      stack_trace: "java.lang.NullPointerException: Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.noText()\"\
        \ because \"inputDataType\" is null\n\tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.rangeTypesCompatible(EnrichLookupService.java:139)\n\
        \tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.validateTypes(EnrichLookupService.java:128)\n\
        \tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:103)\n\
        \tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:63)\n\
        \tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService.doLookup(AbstractLookupService.java:302)\n\
        \tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:440)\n\
        \tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:435)\n\
        \tat org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:76)\n\
        \tat org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:289)\n\
        \tat org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:302)\n\
        \tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)\n\
        \tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\
        \tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)\n\
        \tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)\n\
        \tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\
        \tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)\n\
        \tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)\n\
        \tat java.lang.Thread.run(Thread.java:1447)\n"
```

--- a/docs/changelog/128260.yaml
+++ b/docs/changelog/128260.yaml
@@ -1,0 +1,7 @@
+pr: 128260
+summary: Fix validation NPE in Enrich and add extra @Nullable annotations
+area: ES|QL
+type: bug
+issues:
+ - 126297
+ - 126253

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
@@ -182,7 +182,7 @@ public abstract class AbstractLookupService<R extends AbstractLookupService.Requ
         T request,
         SearchExecutionContext context,
         Block inputBlock,
-        DataType inputDataType,
+        @Nullable DataType inputDataType,
         Warnings warnings
     );
 
@@ -200,7 +200,7 @@ public abstract class AbstractLookupService<R extends AbstractLookupService.Requ
         MappedFieldType field,
         SearchExecutionContext searchExecutionContext,
         Block block,
-        DataType inputDataType
+        @Nullable DataType inputDataType
     ) {
         return switch (inputDataType) {
             case IP -> QueryList.ipTermQueryList(field, searchExecutionContext, (BytesRefBlock) block);

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java
@@ -24,6 +24,7 @@ import org.elasticsearch.compute.data.BlockStreamInput;
 import org.elasticsearch.compute.data.Page;
 import org.elasticsearch.compute.operator.Warnings;
 import org.elasticsearch.compute.operator.lookup.QueryList;
+import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.Releasables;
 import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.index.mapper.RangeFieldMapper;
@@ -103,7 +104,7 @@ public class EnrichLookupService extends AbstractLookupService<EnrichLookupServi
         TransportRequest request,
         SearchExecutionContext context,
         Block inputBlock,
-        DataType inputDataType,
+        @Nullable DataType inputDataType,
         Warnings warnings
     ) {
         MappedFieldType fieldType = context.getFieldType(request.matchField);
@@ -128,8 +129,8 @@ public class EnrichLookupService extends AbstractLookupService<EnrichLookupServi
         return new LookupResponse(in, blockFactory);
     }
 
-    private static void validateTypes(DataType inputDataType, MappedFieldType fieldType) {
-        if (fieldType instanceof RangeFieldMapper.RangeFieldType rangeType) {
+    private static void validateTypes(@Nullable DataType inputDataType, MappedFieldType fieldType) {
+        if (fieldType instanceof RangeFieldMapper.RangeFieldType rangeType && inputDataType != null) {
             // For range policy types, the ENRICH index field type will be one of a list of supported range types,
             // which need to match the input data type (eg. ip-range -> ip, date-range -> date, etc.)
             if (rangeTypesCompatible(rangeType.rangeType(), inputDataType) == false) {
@@ -142,7 +143,7 @@ public class EnrichLookupService extends AbstractLookupService<EnrichLookupServi
         // For geo_match, type validation is done earlier, in the Analyzer.
     }
 
-    private static boolean rangeTypesCompatible(RangeType rangeType, DataType inputDataType) {
+    private static boolean rangeTypesCompatible(RangeType rangeType, @Nullable DataType inputDataType) {
         if (inputDataType.noText() == DataType.KEYWORD) {
             // We allow runtime parsing of string types to numeric types
             return true;

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java
@@ -19,6 +19,7 @@ import org.elasticsearch.compute.data.BlockStreamInput;
 import org.elasticsearch.compute.data.Page;
 import org.elasticsearch.compute.operator.Warnings;
 import org.elasticsearch.compute.operator.lookup.QueryList;
+import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.Releasables;
 import org.elasticsearch.index.query.SearchExecutionContext;
 import org.elasticsearch.index.shard.ShardId;
@@ -83,7 +84,7 @@ public class LookupFromIndexService extends AbstractLookupService<LookupFromInde
         TransportRequest request,
         SearchExecutionContext context,
         Block inputBlock,
-        DataType inputDataType,
+        @Nullable DataType inputDataType,
         Warnings warnings
     ) {
         return termQueryList(context.getFieldType(request.matchField), context, inputBlock, inputDataType).onlySingleValues(

